wingui/pdcscrn.c

```diff
@@ -2252,7 +2254,7 @@ INLINE int set_up_window( void)
     SetTimer( PDC_hWnd, TIMER_ID_FOR_BLINKING, 500, NULL);
     debug_printf( "timer set\n");
 
-#ifdef MONITOR_DEFAULTTONEAREST
+#if WINVER > 0x0500
```

Comment:
Generally speaking,  I'd prefer the conditional compile to depend on something related to the actual code,  rather than to a (possibly related) `WINVER` flag.  I could see issues with different compilers failing to respond properly.

If you're actually seeing an instance where `MONITOR_DEFAULTTONEAREST` is defined,  but there are problems later on with,  say,  `GetMonitorInfo()` _not_ being defined,  then I'd go with this patch.  (Otherwise,  it would seem to be a fix for a non-problem.) Are you seeing failures caused by the existing test?

To deal with possible issues on other compilers,  I'd be apt to make it
```
#if WINVER > 0x0500 && defined( MONITOR_DEFAULTTONEAREST)
```
...i.e.,  we only use this (not entirely necessary) code if we're reasonably certain that all the bits and bobs we're going to need for it are present.

Reply:
you are right but It's very unlikely, I'll add it anyway

Reply:
what about
```c
#if WINVER <= 0x0500
#undef MONITOR_DEFAULTTONEAREST
#endif
```

before
```c
#ifdef MONITOR_DEFAULTTONEAREST
```

Reply:
@Bill-Gray: The problem here is that some compilers (or more specifically, their headers) may define things such as `MONITOR_DEFAULTTONEAREST` regardless of the target Windows version one has specified using macros such as `WINVER` or `_WIN32_WINNT`. An example is MinGW-w64; while it can compile programs for older Windows versions provided they don't use certain CRT features (with the number of problematic features having grown over time), it officially only supports Windows XP and later as a target and therefore its `winuser.h` implementation knows nothing of the likes of Windows 95 or NT 4.0. (I wish the solution here was as simple as "use mainline MinGW", but sadly, it isn't.)

As such, it would be more reliable to check the value of the `WINVER` macro (or similar), ensuring the desired results regardless of the compiler used.

By the way, it would be better to use `#if WINVER >= 0x0410` since, apparently, these multi-monitor functions were added in Windows 98. Or something such as `#if (_WIN32_WINNT >= 0x0500) || (_WIN32_WINDOWS >= 0x0410)`.

Reply:
unfortunately msdn does not reports anymore compatibility of api on old systems, so the only way is check in a vm, you use `MONITOR_DEFAULTTONEAREST` as check, undefining it looks a good idea to me, regardless of the ifdef involved (that needs some investigation, I also have a win98 vm)